vioscsi/helper.c

```diff
@@ -109,7 +109,7 @@ ENTER_FN_SRB();
         srbExt->out, srbExt->in,
         &srbExt->cmd, va, pa);
 
-    if (res >= 0) {
+    if (res == VQ_ADD_BUFFER_SUCCESS) {
```

Comment:
both the split and packed variant either return -ENOSPC or zero, which made me think - I don't think ENOSPC is technically the right thing to do here. 

I'd imagine we should actually use ENOMEM or ENOBUFS, as that describes what is happening better than ENOSPC, which is supposed to be used for disk space, no? (ref https://www.gnu.org/software/libc/manual/html_node/Error-Codes.html)

Reply:
Although usually applying to disk devices, ENOSPC refers to _any_ device. The mnemonic ENOSPC refers to ERROR_NO_SPACE. AFAIK, this is returned from the hypervisor without translation rather than something we set. Typically ENOMEM == ENOBUFS in GNU (per your link) and these relate to allocation of virtual memory, whereas virtio sees a _device_ responding with "I have no space for that".

Reply:
Per my comment below, we do actually set `-ENOSPC`, e.g.:

https://github.com/virtio-win/kvm-guest-drivers-windows/blob/22d0908a28f382d02d8d3a10a0ee0631af480fdf/VirtIO/VirtIORing.c#L263